metals/src/main/scala/scala/meta/internal/metals/SemanticdbIndexer.scala

```diff
@@ -39,6 +40,20 @@ class SemanticdbIndexer(
     path.semanticdbRoot.foreach(onChangeDirectory(_))
   }
 
+  /**
+   * Handle EventType.OVERFLOW, meaning we lost file events, when we don't know the path.
+   * We walk up the file tree for all targets `META-INF/semanticdb` directory
+   * and re-index all of its `*.semanticdb` children.
+   */
+  def onOverflow(): Unit = {
+    buildTargets.scalacOptions.foreach { item =>
+      val targetroot = item.targetroot
+      if (!targetroot.isJar) {
+        onChangeDirectory(targetroot.resolve(Directories.semanticdb).toNIO)
```

Comment:
Something to take into consideration, we use FixedThreadPool(1) to handle events, because reindexing is sync, we will stop processing events till reindexing is done.

Reply:
changed to using future there